HealthCheck: COMMAND health check without port is not possible

I made an app definition without port definition and a command health check.
```json
 "ports": [ ],
  "healthChecks": [
    {
      "protocol": "COMMAND",
      "command": { "value": "testHealth.sh" },
      "gracePeriodSeconds": 120,
      "intervalSeconds": 60,
      "maxConsecutiveFailures": 3
    }
  ]
```
Deployment is not possible and gives this error:
```
java.lang.AssertionError: assertion failed: Health check port indices must address an element of the ports array or container port mappings.
```
To work around this issue, I added a port in the ports array.

The above definition should be possible without a port definition.

--- a/src/main/scala/mesosphere/marathon/state/AppDefinition.scala
+++ b/src/main/scala/mesosphere/marathon/state/AppDefinition.scala
@@ -11,6 +11,7 @@ import mesosphere.marathon.health.HealthCheck
 import mesosphere.marathon.state.Container.Docker.PortMapping
 import mesosphere.marathon.state.PathId._
 import mesosphere.marathon.Protos
+import mesosphere.marathon.Protos.HealthCheckDefinition.Protocol
 import mesosphere.marathon.upgrade.DeploymentPlan
 import mesosphere.mesos.TaskBuilder
 import mesosphere.mesos.protos.{ Resource, ScalarResource }
@@ -85,7 +86,7 @@ case class AppDefinition(
   def portIndicesAreValid(): Boolean = {
     val validPortIndices = 0 until hostPorts.size
     healthChecks.forall { hc =>
-      validPortIndices contains hc.portIndex
+      hc.protocol == Protocol.COMMAND || (validPortIndices contains hc.portIndex)
     }
   }
 

--- a/src/test/scala/mesosphere/marathon/state/AppDefinitionTest.scala
+++ b/src/test/scala/mesosphere/marathon/state/AppDefinitionTest.scala
@@ -6,6 +6,7 @@ import com.google.common.collect.Lists
 import mesosphere.marathon.upgrade.DeploymentPlan
 import mesosphere.marathon.{ Protos, MarathonSpec }
 import mesosphere.marathon.Protos.{ Constraint, ServiceDefinition }
+import mesosphere.marathon.Protos.HealthCheckDefinition.Protocol
 import mesosphere.marathon.api.ModelValidation
 import mesosphere.marathon.api.v2.json.EnrichedTask
 import mesosphere.marathon.health.HealthCheck
@@ -223,6 +224,20 @@ class AppDefinitionTest extends MarathonSpec with Matchers with ModelValidation 
       "Health check port indices must address an element of the ports array or container port mappings."
     )
 
+    shouldNotViolate(
+      correct.copy(
+        container = Some(Container(
+          docker = Some(Docker(
+            network = Some(mesos.ContainerInfo.DockerInfo.Network.BRIDGE),
+            portMappings = None
+          ))
+        )),
+        ports = Nil,
+        healthChecks = Set(HealthCheck(protocol = Protocol.COMMAND))
+      ),
+      "",
+      "Health check port indices must address an element of the ports array or container port mappings."
+    )
   }
 
   test("SerializationRoundtrip") {

